pages/album/[id].js

```diff
@@ -1,30 +1,56 @@
 import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
 import AlbumDetail from "../../components/AlbumDetail/AlbumDetail.js";
-import curtisAlbum from "../../public/images/curtis.jpeg";
 import AppHeader from "../../components/Header/Header.js";
 
 export default function AlbumDetailPage() {
   const router = useRouter();
-  const { id } = router.query;
+  const { id, coverImage } = router.query;
+  const [album, setAlbum] = useState(null);
 
-  const album = {
-    name: "Curtis Mayfield",
-    title: "Curtis (Deluxe Edition)",
-    genre: "Funk, Soul",
-    year: 1970,
-    id: id,
-    src: curtisAlbum,
-  };
+  useEffect(() => {
+    async function fetchAlbum() {
+      try {
+        const response = await fetch(`https://api.discogs.com/releases/${id}`);
```

Comment:
`data.images[0].resource_url` **will always be an empty string as long as you are not authenticated with the discogs api**.

To do so you need to move this into your own api route:

e.g. `pages/api/releases/[id].js`

Your file could look something like this (code is not tested):
```js
export default async function handler(request, response) {
  const { id } = request.query;

  const headers = {
    Authorization: `Discogs token=${process.env.DISCOGS_TOKEN}`,
  };

  try {
    const fetchResponse = await fetch(`https://api.discogs.com/releases/${id}`, {
      headers,
    });
    const data = await fetchResponse.json();
    response.status(200).json(data);
  } catch (error) {
    response.status(error.response.status).json({ message: error.response.statusText });
  }
}
```

Then instead of the official api you can fetch your own api proxy:

```suggestion
        const response = await fetch(`/api/releases/${id}`);
```

This endpoint should then include all images.

Reply:
Of course you need to add `DISCOGS_TOKEN` as an environment variable to you `.env.local` file as well as to the vercel project settings.